### PR TITLE
feat: support iteration over list values that automatically fetches next

### PR DESCRIPTION
**Pull request by lhjnilsson**

### CODE REVIEW

File: lemon/market_data/instruments.py [hunk 1]
```diff
@@ -36,3 +36,36 @@ def get(
...
+                    "sorting": sorting,
+                    "limit": limit,
+                    "page": page,
+                },
+            )
+            for result in resp.json()["results"]:
+                yield Instrument._from_data(result)
+            if resp.json()["page"] == resp.json()["pages"]:
+                break
+            page = int(resp.json()["page"]) + 1
```
Comment (dnnsthnnr) [inline lemon/market_data/instruments.py:71]: We are already outputting the next link in the response, I would directly use it.

Comment (lhjnilsson) [reply lemon/market_data/instruments.py:71]: Then we need to redo the client. Client currently holds the BASE_URL and expects part of the url such as "instruments"

Comment (dynox) [reply lemon/market_data/instruments.py:71]: Agree with Dennis statement, we don't have to rewrite the client too cause urljoin should take care about proper url concatenation like in picture below

image

Comment (lhjnilsson) [reply lemon/market_data/instruments.py:71]: Did not know this! will do it then.
However, to the most important note. This aproach to implement a iter is feasible for us? so i can implement for the other iteratable- endpoints

**Review by dnnsthnnr — CHANGES_REQUESTED** [review]
LGTM,
but let's add an example to the readme and also point out that the iteration is blocking and respecting the rate limit of the user. Once hit the iterator waits until the rate limit resets and then requests the next pages.

File: lemon/trading/account.py [hunk 2]
```diff
@@ -110,6 +156,29 @@ def get_documents(
         )
         return GetDocumentsResponse._from_data(resp.json())
 
+    def iter_document(
+        self,
+        sorting: Optional[Sorting] = None,
+        limit: Optional[int] = None,
+        page: Optional[int] = None,
```
Comment (lhjnilsson) [inline lemon/trading/account.py:163]: remove this, page does not make sense when iter

**Review by dynox — CHANGES_REQUESTED** [review]
We could go with solution similar to what can be seen [here](https://stripe.com/docs/api/pagination/auto?lang=python).

The basic usage would be to:
1. call the endpoint using regular endpoint call, getting the paginated response object
2. bind client with paginated response object (client shouldn't be bounded to every response type, only to those which can be paginated)
3. iterate until series is exhausted

This approach is cleaner than proposed solution. We could apply it everywhere when needed without adding to much changes.

**Pull request closed without merging**